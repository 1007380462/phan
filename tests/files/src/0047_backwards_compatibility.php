<?php
echo $foo->$bar['baz'];
Foo::$bar['baz']();
$foo->$bar['baz']();
strlen($foo->$bar['baz']);

// tests that should pass without warning below
class Test {
    public static $vals = array('a' => 'A', 'b' => 'B');
    public static function get($letter) {
        return self::$vals[$letter];
    }
}
