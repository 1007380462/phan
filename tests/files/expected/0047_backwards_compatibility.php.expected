%s:2 CompatError Expression may not be PHP 7 compatible
%s:3 CompatError Expression may not be PHP 7 compatible
%s:4 CompatError Expression may not be PHP 7 compatible
%s:5 CompatError Expression may not be PHP 7 compatible
